analyzers/sdk_test.go

```diff
@@ -0,0 +1,170 @@
+package analyzers
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+
+	"github.com/BurntSushi/toml"
+	"github.com/deepsourcelabs/deepsource-go/analyzers/analysistest"
+	"github.com/deepsourcelabs/deepsource-go/analyzers/processors"
+	"github.com/deepsourcelabs/deepsource-go/analyzers/types"
+	"github.com/deepsourcelabs/deepsource-go/analyzers/utils"
+)
+
+func TestAnalyzer(t *testing.T) {
+	t.Run("Run staticcheck as DeepSource Analyzer", func(t *testing.T) {
+		// set environment variables
+		tempDir := t.TempDir()
+		t.Setenv("TOOLBOX_PATH", tempDir)
+		t.Setenv("REPO_ROOT", tempDir)
+
+		rp := processors.RegexProcessor{
+			Pattern: `(?P<filename>.+):(?P<line>\d+):(?P<column>\d+): (?P<message>.+)\((?P<issue_code>\w+)\)`,
+		}
+
+		a := &CLIRunner{
+			Name:      "staticcheck",
+			Command:   "staticcheck",
+			Args:      []string{"-f", "text", "./testdata/src/staticcheck/..."},
+			Processor: &rp,
+		}
+
+		err := testRunner(a, tempDir, "testdata/src/staticcheck/staticcheck.go")
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	t.Run("Run csslint as DeepSource Analyzer", func(t *testing.T) {
+		// set environment variables
+		tempDir := t.TempDir()
+		t.Setenv("TOOLBOX_PATH", tempDir)
+		t.Setenv("REPO_ROOT", tempDir)
+
+		issueProcessor := func(content string) string {
+			issueMap := map[string]string{
+				"empty-rules":      "E001",
+				"errors":           "E002",
+				"known-properties": "K001",
+			}
+
+			if issueMap[content] == "" {
+				return "U001"
+			}
+
+			return issueMap[content]
+		}
+
+		rp := processors.RegexProcessor{
+			Pattern:            `(?P<filename>.+): line (?P<line>\d+), col (?P<column>\d+), (?P<message>.+) \((?P<issue_code>.+)\)`,
+			IssueCodeProcessor: issueProcessor,
+		}
+
+		a := &CLIRunner{
+			Name:      "csslint",
+			Command:   "csslint",
+			Args:      []string{"--format=compact", "."},
+			Processor: &rp,
+		}
+
+		err := testRunner(a, tempDir, "testdata/src/csslint/csslint.css")
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+}
+
+func testRunner(a *CLIRunner, tempDir string, triggerFilename string) error {
```

Comment:
Maybe put this function in the `analysistest` package, and change the signature to `Run(a *CLIRunner, testDir string) error`?

The developer can call this to test their analyzer, so they need not worry about the `tempDir`. The `Run` function should take care of that.